src/functions.php

```diff
@@ -25,15 +25,29 @@ function timeout(PromiseInterface $promise, $time, LoopInterface $loop)
             $loop->cancelTimer($timer);
             $reject($v);
         });
+    }, function ($resolveUnused, $reject) use ($loop, &$timer, $promise) {
+        // cancelling this promise will cancel the timer and reject
+        $loop->cancelTimer($timer);
+        $reject(new \RuntimeException('Timeout cancelled'));
```

Comment:
> the default cancelling behaviour of the input promise

I'm not sure I follow, but how can we rely on its handling?

It's my understanding that the input promise *can* support cancellation, but we can't really rely on its outcome.
Also, I'm not sure about the rejection value?